app/controllers/mixins/actions/vm_actions/reconfigure.rb

```diff
@@ -19,7 +19,8 @@ def reconfigure(reconfigure_ids = [])
 
           if @reconfigitems.size == 1
             vm = Vm.find(@reconfigitems)
-            @vlan_options = get_vlan_options(vm.host_id)
+            @vlan_options = vm.try(:available_networks) || get_vlan_options(vm.host_id)
```

Comment:
get_vlan_options is only called from this place and returns a list of Lan objects

with this change, you try to call a method available only on one provider and returns list of CloudNetwork objects (not related to Lan)

So - why would you put that try here instead of inside the function written specifically to give you the available lans?

And, why does it make sense to use the same field for 2 unrelated uses?

Reply:
Sorry for pissing you off 😃 My thinking was that "Lan" is in fact infra version of "Private Network" for cloud providers, hence I joined them in same field. Didn't reuse `get_vlan_options` function because it relies only on `vm.host_id` which is always `nil` for our cloud provider. But you're right, let me isolate the drop-down from exisiting one completely to improve readability.

Reply:
Thanks... I mean, if mixing lans and networks makes sense from the provider perspective, by all means feel free to do so.

I just see Lan & CloudNetwork being mixed in one field and it feel like that's probably a bug :).

Reply:
@himdel is it worth investing time in angular drop-down instead rails? If yes, I kindly ask for pointer to example.

Reply:
Uh, what would be the difference?

Do you mean using `options_for_select` in the template, versus retrieving the options via the API and populating via `ng-options`?

Reply:
That's what I meant, yes, but I'm not sure anymore why would I want to do it this way :)

Reply:
Then I would not worry about it too much :)

But if you want.. the idea is that we should be preferring to use the API whenever possible (so that we can remove the UI code which duplicates what the API does).

So.. the select would look like this:
https://github.com/ManageIQ/manageiq-ui-classic/pull/3739/files#diff-31ab4a7ef602ebbddb2e18bde4188f6dR44

And getting the data somewhat like (except not in miqService so you don't have to support custom callbacks :)):
https://github.com/ManageIQ/manageiq-ui-classic/pull/3739/files#diff-e27d4b8817a8b9402efc6a4e6800a565R184


So.. if the API already returns the right thing, deleting the `vlan_options` code is definitely preferred to adding a special case for it :). If not, meh, would be nice.

Reply:
I've done as you suggest, thanks for the links. I'm only struggling with empty select value that somehow appears:

![capture](https://user-images.githubusercontent.com/8102426/40619154-47b2440e-6294-11e8-9580-094613b256a4.PNG)

Do you happen to know how to get rig of it?

```haml
              %select{'name'                    => 'adapterNetwork',
                'ng-model'                    => 'vm.reconfigureModel.adapterNetwork',
                'ng-options'                  => 'network for network in vm.reconfigureModel.availableAdapterNetworks',
                'pf-select'                   => true,
                'required'                    => false,
                'selectpicker-for-select-tag' => ''}
              %option{"value" => ""}
                = "<#{_('Choose')}>"
```